store/newProduct.js

```diff
@@ -0,0 +1,81 @@
+function newVariety() {
+  return {
+    color: null,
+    sizes: {
+      XS:  0,
+      S:   0,
+      M:   0,
+      L:   0,
+      XL:  0,
+      XXL: 0,
+    },
+    quantity: 0,
+    images: [],
+  };
+}
+
+export default {
+  namespaced: true,
+  state() {
+    const initialSizes = false;
+    return {
+      name: '',
+      type: '',
+      description: '',
+      price: 0,
+      inSizes: initialSizes,
+      quantity: 0,
+      varieties: [newVariety()],
+      varietiesCount: 1,
+    };
+  },
+  mutations: {
+    reset(state) {
+      const initialSizes = false;
+      state.name = '';
+      state.type = '';
+      state.description = '';
+      state.price = 0;
+      state.inSizes = initialSizes;
+      if(initialSizes) state.quantity = 0;
+      state.varieties.forEach(v => v.images.splice(0, v.images.length));  // to clear the old references
+      state.varieties = [newVariety()];
+      state.varietiesCount = 1;
+    },
+    setName(state, payload) {
```

Comment:
create a function that accepts name of the field and gives it a payload value

```suggestion
    setField(state, {name, value}) {
      state[name] = value;
    }
```